### PR TITLE
Auto-advance

### PR DESCRIPTION
**Pull request by NicolasBourdin88**
Depends on #1850

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -396,6 +396,17 @@
     <string name="settingsAiEngineDescription">Select your AI writing assistant</string>
     <string name="settingsAppLock">App lock</string>
     <string name="settingsAuthorizeTracking">Authorize tracking</string>
+    <string name="settingsAutoAdvanceDescription">Select what to display when a message is archived or deleted\u0020</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values/strings.xml:399]: `\u0020` ?

Remove it on Loco.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -396,6 +396,17 @@
...
+    <string name="settingsAutoAdvanceFollowingThreadDescription">Display following thread when archiving or deleting\n</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values/strings.xml:400]: `\n` ?

Remove it on Loco, on all locales.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt [hunk 3]
```diff
@@ -71,6 +71,11 @@ class ThreadController @Inject constructor(
         return getThreadQuery(uid, mailboxContentRealm()).asFlow()
     }
 
+    fun getMessageCountInThreadForFolder(threadUid: String, folderId : String) : Int{
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt:74]: Auto-format.

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt [hunk 4]
```diff
@@ -71,6 +71,11 @@ class ThreadController @Inject constructor(
...
+        return getThreadQuery(threadUid, mailboxContentRealm()).find()?.messages?.query("${Message::folderId.name} == $0", folderId)?.count()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt:75]: Is the 1st `find()` necessary ? I'm in doubt.

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt (hunk 3 above)
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/ThreadController.kt:74]: This function is getting data about Messages, so move it to the MessageController.

File: app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt [hunk 5]
```diff
@@ -191,6 +193,13 @@ class LocalSettings private constructor(context: Context) : SharedValues {
         ASK_ME("false", R.string.settingsOptionAskMe, "askMe"),
     }
 
+    enum class AutoAdvanceMode(val id: String, @StringRes val localisedNameRes: Int) {
+        PREVIOUS_THREAD("previousThread", R.string.settingsAutoAdvancePreviousThreadDescription),
+        FOLLOWING_THREAD("followingThread", R.string.settingsAutoAdvanceFollowingThreadDescription),
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt:198]: Check with iOS if previous/following mean the same for both OS.

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 6]
```diff
@@ -499,6 +506,7 @@ class MainViewModel @Inject constructor(
             undoDestinationId,
             threads.count(),
         )
+
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:509]:
```suggestion
```

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 7]
```diff
@@ -643,6 +659,7 @@ class MainViewModel @Inject constructor(
                 destinationFolderId = destinationFolder.id,
                 callbacks = RefreshCallbacks(::onDownloadStart, ::onDownloadStop),
             )
+
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:662]:
```suggestion
```

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 8]
```diff
@@ -1039,6 +1056,15 @@ class MainViewModel @Inject constructor(
...
     }
 
+    private fun isMessageTheLastInFolder(threadUid: String): Boolean {
+        currentFolderId?.let { folderId ->
+            if (threadController.getMessageCountInThreadForFolder(threadUid, folderId) == 1) {
+                return true
+            }
+        }
+        return false
+    }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:1066]:
```suggestion
    private fun isMessageTheLastInFolder(threadUid: String): Boolean {
        val folderId = currentFolderId ?: return false
        return threadController.getMessageCountInThreadForFolder(threadUid, folderId) == 1
    }
```

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 9]
```diff
@@ -1039,6 +1056,15 @@ class MainViewModel @Inject constructor(
         snackbarManager.postValue(appContext.getString(snackbarTitleRes))
     }
 
+    private fun isMessageTheLastInFolder(threadUid: String): Boolean {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:1059]: The name is confusing.

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 10]
```diff
@@ -559,7 +616,7 @@ class ThreadListAdapter @Inject constructor(
         return getItemViewType(position) == DisplayType.THREAD.layout && swipingIsAuthorized
     }
 
-    override fun createDiffUtil(oldList: List<Any>, newList: List<Any>): DragDropSwipeDiffCallback<Any>? = null
+    override fun createDiffUtil(oldList: List<Any>, newList: List<Any>) = null
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:619]: Why this change ?

File: app/src/main/java/com/infomaniak/mail/ui/main/settings/AutoAdvanceSettingsFragment.kt [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return FragmentAutoAdvanceSettingsBinding.inflate(inflater, container, false).also { binding = it }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) = with(binding) {
+        super.onViewCreated(view, savedInstanceState)
+        setSystemBarsColors()
+
+        radioGroup.initBijectionTable(
+            R.id.lastAction to LocalSettings.AutoAdvanceMode.NATURAL_THREAD,
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/settings/AutoAdvanceSettingsFragment.kt:51]: Import `LocalSettings.`.

File: app/src/main/java/com/infomaniak/mail/ui/main/settings/AutoAdvanceSettingsFragment.kt [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+            R.id.nextThread to LocalSettings.AutoAdvanceMode.FOLLOWING_THREAD,
+            R.id.listThread to LocalSettings.AutoAdvanceMode.LIST_THREAD,
+            R.id.lastThread to LocalSettings.AutoAdvanceMode.PREVIOUS_THREAD,
+        )
+
+        radioGroup.check(localSettings.autoAdvanceMode)
+
+        radioGroup.onItemCheckedListener { _, _, autoAdvanceMode ->
+            chooseAutoAdvanceMode(autoAdvanceMode as LocalSettings.AutoAdvanceMode)
+            lastAction.setSubTextVisibility(if (autoAdvanceMode == LocalSettings.AutoAdvanceMode.NATURAL_THREAD) View.VISIBLE else View.GONE)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/settings/AutoAdvanceSettingsFragment.kt:61]: Use `.isVisible` instead.

File: app/src/main/java/com/infomaniak/mail/ui/main/settings/SettingRadioButtonView.kt [hunk 13]
```diff
@@ -55,9 +57,19 @@ class SettingRadioButtonView @JvmOverloads constructor(
...
 
+                getString(R.styleable.SettingRadioButtonView_subText).let {
+                    subText.apply {
+                        subText.text = it
+                        subText.visibility = GONE
+                    }
+                }
+
+
+
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/settings/SettingRadioButtonView.kt:72]:
```suggestion
```

File: app/src/main/res/layout/fragment_settings.xml [hunk 14]
```diff
@@ -102,6 +102,15 @@
...
 
+            <com.infomaniak.mail.ui.main.settings.ItemSettingView
+                android:id="@+id/settingsAutomaticAdvance"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                app:itemAction="chevron"
+                app:title="@string/settingsAutoAdvanceTitle"
+                tools:subtitle="@string/settingsAutoAdvanceDescription" />
+
+
```
Comment (KevinBoulongne) [inline app/src/main/res/layout/fragment_settings.xml:113]:
```suggestion
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt (hunk 5 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/data/LocalSettings.kt:198]: Democracy won.
We need to swap this.

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt (hunk 9 above)
Comment (NicolasBourdin88) [reply app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:1059]: is this name better : threadHasOnlyOneMessage

File: app/src/main/res/layout/fragment_settings.xml [hunk 15]
```diff
@@ -102,6 +102,15 @@
                 app:title="@string/settingsExternalContentTitle"
                 tools:subtitle="@string/settingsOptionAskMe" />
 
+            <com.infomaniak.mail.ui.main.settings.ItemSettingView
+                android:id="@+id/settingsAutomaticAdvance"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                app:itemAction="chevron"
+                app:title="@string/settingsAutoAdvanceTitle"
+                tools:subtitle="@string/settingsAutoAdvanceDescription" />
```
Comment (LunarX) [inline app/src/main/res/layout/fragment_settings.xml:111]: I don't think this the tools subtitle is correct, it should show an example of a selected value instead

File: app/src/main/res/layout/view_setting_radio_button.xml [hunk 16]
```diff
@@ -33,12 +33,28 @@
...
+
+        <TextView
+            android:id="@+id/subText"
+            style="@style/BodySmall.Tertiary"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:visibility="gone"
+            tools:text="@string/settingsDefault"
+            tools:visibility="visible"
+            android:paddingTop="@dimen/marginStandardVerySmall"/>
```
Comment (LunarX) [inline app/src/main/res/layout/view_setting_radio_button.xml:56]: Same comment as for the class, you don't need to put this text inside the SettingRadioButtonView, you can add a separate simple TextView outside the item

File: app/src/main/res/navigation/main_navigation.xml [hunk 17]
```diff
@@ -309,8 +306,23 @@
         <action
             android:id="@+id/action_settings_to_dataManagementSettings"
             app:destination="@id/dataManagementSettingsFragment" />
+        <action
+            android:id="@+id/action_settings_to_externalContentSetting"
+            app:destination="@id/externalContentSettingFragment" />
```
Comment (LunarX) [inline app/src/main/res/navigation/main_navigation.xml:311]: This direction was better placed before you moved it

File: app/src/main/res/navigation/main_navigation.xml [hunk 18]
```diff
@@ -309,8 +306,23 @@
...
+        <action
+            android:id="@+id/action_settingsFragment_to_aiEngineSettingFragment"
+            app:destination="@id/aiEngineSettingFragment" />
```
Comment (LunarX) [inline app/src/main/res/navigation/main_navigation.xml:314]: This direction already exists and is better placed, so remove the one that's been added with your PR

File: app/src/main/res/navigation/main_navigation.xml [hunk 19]
```diff
@@ -309,8 +306,23 @@
...
+        <action
+            android:id="@+id/action_settings_to_autoAdvanceSettings"
+            app:destination="@id/autoAdvanceFragment" />
```
Comment (LunarX) [inline app/src/main/res/navigation/main_navigation.xml:317]: You can move this direction right after the original external content direction

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (LunarX) [inline app/src/main/res/values/strings.xml:399]: There's an empty whitespace at the end of your string

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (LunarX) [inline app/src/main/res/values/strings.xml:400]: There's a useless new line at the end of this translation. It also applies to other languages

File: app/src/main/res/values-es/strings.xml [hunk 20]
```diff
@@ -390,6 +390,17 @@
     <string name="settingsAiEngineDescription">Seleccione su asistente de redacción</string>
     <string name="settingsAppLock">Bloqueo de aplicaciones</string>
     <string name="settingsAuthorizeTracking">Autorizar el seguimiento</string>
+    <string name="settingsAutoAdvanceDescription">Seleccione qué mostrar cuando se archiva o elimina un mensaje</string>
+    <string name="settingsAutoAdvanceFollowingThreadDescription">Mostrar el siguiente hilo al archivar o borrar\n</string>
+    <string name="settingsAutoAdvanceFollowingThreadTitle">Siguiendo el hilo</string>
+    <string name="settingsAutoAdvanceListOfThreadsDescription">Mostrar lista de hilos al archivar o borrar</string>
+    <string name="settingsAutoAdvanceListOfThreadsTitle">Lista de temas</string>
```
Comment (LunarX) [inline app/src/main/res/values-es/strings.xml:397]: The term "conversation" is usually not translated as "temas" in spanish translations, maybe check if you can find synonyms that are already used in other already existing translations like "conversación"

File: app/src/main/res/values-es/strings.xml [hunk 21]
```diff
@@ -390,6 +390,17 @@
...
+    <string name="settingsAutoAdvanceNaturalThreadDescription">Repite la última acción realizada</string>
+    <string name="settingsAutoAdvanceNaturalThreadHint">Selecciona automáticamente la conversación siguiente o anterior, en función de tu última interacción.</string>
+    <string name="settingsAutoAdvanceNaturalThreadTitle">Repetir la última acción</string>
+    <string name="settingsAutoAdvancePreviousThreadDescription">Mostrar el hilo anterior al archivar o borrar</string>
+    <string name="settingsAutoAdvancePreviousThreadTitle">Tema anterior</string>
```
Comment (LunarX) [inline app/src/main/res/values-es/strings.xml:402]: Same comment for "Tema"

File: app/src/main/res/values-es/strings.xml [hunk 22]
```diff
@@ -390,6 +390,17 @@
...
+    <string name="settingsAutoAdvanceTitle">Auto advance</string>
```
Comment (LunarX) [inline app/src/main/res/values-es/strings.xml:403]: Not sure this one has been translated correctly

File: app/src/main/res/values-es/strings.xml [hunk 23]
```diff
@@ -390,6 +390,17 @@
...
     <string name="settingsAuthorizeTracking">Autorizar el seguimiento</string>
+    <string name="settingsAutoAdvanceDescription">Seleccione qué mostrar cuando se archiva o elimina un mensaje</string>
+    <string name="settingsAutoAdvanceFollowingThreadDescription">Mostrar el siguiente hilo al archivar o borrar\n</string>
+    <string name="settingsAutoAdvanceFollowingThreadTitle">Siguiendo el hilo</string>
+    <string name="settingsAutoAdvanceListOfThreadsDescription">Mostrar lista de hilos al archivar o borrar</string>
+    <string name="settingsAutoAdvanceListOfThreadsTitle">Lista de temas</string>
+    <string name="settingsAutoAdvanceNaturalThreadDescription">Repite la última acción realizada</string>
+    <string name="settingsAutoAdvanceNaturalThreadHint">Selecciona automáticamente la conversación siguiente o anterior, en función de tu última interacción.</string>
+    <string name="settingsAutoAdvanceNaturalThreadTitle">Repetir la última acción</string>
+    <string name="settingsAutoAdvancePreviousThreadDescription">Mostrar el hilo anterior al archivar o borrar</string>
```
Comment (LunarX) [inline app/src/main/res/values-es/strings.xml:401]: Try and replace all the usages of "hilo" that you added with "conversación" -> https://fr.pons.com/traduction/espagnol-fran%C3%A7ais/hilo

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 24]
```diff
@@ -289,6 +293,16 @@ class ThreadListAdapter @Inject constructor(
         if (multiSelection?.isEnabled == true) {
             toggleMultiSelectedThread(thread)
         } else {
+            previousThreadClickedPosition?.let { previousThreadClickedPosition ->
+                if (position > previousThreadClickedPosition) {
+                    localSettings.autoAdvanceIntelligentMode = AutoAdvanceMode.FOLLOWING_THREAD
+                } else {
+                    localSettings.autoAdvanceIntelligentMode = AutoAdvanceMode.PREVIOUS_THREAD
+                }
+            }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:302]:
```suggestion
            previousThreadClickedPosition?.let {
                localSettings.autoAdvanceIntelligentMode = if (position > it) {
                    AutoAdvanceMode.FOLLOWING_THREAD
                } else {
                    AutoAdvanceMode.PREVIOUS_THREAD
                }
            }
```

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/res/values-es/strings.xml [hunk 25]
```diff
@@ -390,6 +390,17 @@
...
+    <string name="settingsAutoAdvanceFollowingThreadDescription">Mostrar el siguiente conversación al archivar o borrar</string>
+    <string name="settingsAutoAdvanceFollowingThreadTitle">Siguiendo el conversación</string>
+    <string name="settingsAutoAdvanceListOfThreadsDescription">Mostrar lista de conversaciones al archivar o borrar</string>
+    <string name="settingsAutoAdvanceListOfThreadsTitle">Lista de conversaciones</string>
+    <string name="settingsAutoAdvanceNaturalThreadDescription">Repite la última acción realizada</string>
+    <string name="settingsAutoAdvanceNaturalThreadHint">Selecciona automáticamente la conversación siguiente o anterior, en función de tu última interacción.</string>
+    <string name="settingsAutoAdvanceNaturalThreadTitle">Repetir la última acción</string>
+    <string name="settingsAutoAdvancePreviousThreadDescription">Mostrar el conversación anterior al archivar o borrar</string>
+    <string name="settingsAutoAdvancePreviousThreadTitle">Conversación anterior</string>
+    <string name="settingsAutoAdvanceTitle">Avance automático</string>
```
Comment (LunarX) [inline app/src/main/res/values-es/strings.xml:403]: conversación is feminine which means some of the translation are wrong. settingsAutoAdvanceFollowingThreadDescription, settingsAutoAdvanceFollowingThreadTitle, settingsAutoAdvancePreviousThreadDescription

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt [hunk 26]
```diff
@@ -137,6 +138,12 @@ class MessageController @Inject constructor(private val mailboxContentRealm: Rea
     fun getMessagesAsync(messageUid: String): Flow<ResultsChange<Message>> {
         return getMessagesQuery(messageUid, mailboxContentRealm()).asFlow()
     }
+
+    fun getMessageCountInThreadForFolder(threadUid: String, folderId: String, realm: Realm): Long? {
+        return ThreadController.getThread(threadUid, realm)
+            ?.messages?.query("${Message::folderId.name} == $0", folderId)
+            ?.count()?.find()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/MessageController.kt:145]:
```suggestion
            ?.count()
            ?.find()
```

File: app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt [hunk 27]
```diff
@@ -462,6 +465,12 @@ class ThreadFragment : Fragment() {
         }
     }
 
+    private fun observeClosedThread() {
+        mainViewModel.autoAdvanceTrigger.observe(viewLifecycleOwner) { listThreadsUids ->
+            tryToAutoAdvance(localSettings.autoAdvanceMode, listThreadsUids)
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadFragment.kt:470]: Don't send `autoAdvanceMode` all the way down from here.
Directly use it where needed.

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 28]
```diff
@@ -110,6 +110,8 @@ class MainViewModel @Inject constructor(
     val reportPhishingTrigger = SingleLiveEvent<Unit>()
     val canInstallUpdate = MutableLiveData(false)
 
+    val autoAdvanceTrigger = MutableLiveData<List<String>>()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:113]: Rename to `autoAdvanceThreadsUids`.

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 29]
```diff
@@ -571,6 +579,8 @@ class MainViewModel @Inject constructor(
         val apiResponse = ApiRepository.moveMessages(mailbox.uuid, messages.getUids(), destinationFolder.id)
 
         if (apiResponse.isSuccess()) {
+            autoAdvanceTrigger.postValue(threadsUids.toList())
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:582]: If you only move 1 Message from a long Thread, why should you auto-advance ?

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 30]
```diff
@@ -307,6 +321,20 @@ class ThreadListAdapter @Inject constructor(
         if (newPosition != null) notifyItemChanged(newPosition, NotificationType.SELECTED_STATE)
     }
 
+    fun getNextThread(startingThreadIndex: Int, direction: Int): Pair<Thread, Int>? {
+        var currentIndexThread = startingThreadIndex
```
[the remark this answers is not in the document]
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:325]: I've had other comments about this function, but finally I just directly rewritten it _(in the next comment)_.

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 31]
```diff
@@ -307,6 +321,20 @@ class ThreadListAdapter @Inject constructor(
...
+        while (currentIndexThread >= 0 && currentIndexThread <= dataSet.lastIndex) {
+            if (dataSet[currentIndexThread] is Thread) {
+                val thread = dataSet[currentIndexThread] as Thread
+                return thread to currentIndexThread
+            }
+
+            currentIndexThread += direction
+        }
+        return null
+    }
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:336]:
```suggestion
    fun getNextThread(startingThreadIndex: Int, direction: Int): Pair<Thread, Int>? {
        var currentThreadIndex = startingThreadIndex + direction

        while (currentThreadIndex >= 0 && currentThreadIndex <= dataSet.lastIndex) {
            if (dataSet[currentThreadIndex] is Thread) return dataSet[currentThreadIndex] as Thread to currentThreadIndex
            currentThreadIndex += direction
        }

        return null
    }
```

File: app/src/main/res/values/strings.xml [hunk 32]
```diff
@@ -396,6 +396,17 @@
...
+    <string name="settingsAutoAdvanceFollowingThreadDescription">Display following thread when archiving or deleting</string>
+    <string name="settingsAutoAdvanceFollowingThreadTitle">Following thread</string>
+    <string name="settingsAutoAdvanceListOfThreadsDescription">Display list of threads when archiving or deleting</string>
+    <string name="settingsAutoAdvanceListOfThreadsTitle">List of threads</string>
+    <string name="settingsAutoAdvanceNaturalThreadDescription">Repeats the last action you performed</string>
+    <string name="settingsAutoAdvanceNaturalThreadHint">Automatically selects the next or previous thread, depending on your last interaction.</string>
+    <string name="settingsAutoAdvanceNaturalThreadTitle">Repeat last action</string>
+    <string name="settingsAutoAdvancePreviousThreadDescription">Display previous thread when archiving or deleting</string>
+    <string name="settingsAutoAdvancePreviousThreadTitle">Previous thread</string>
+    <string name="settingsAutoAdvanceTitle">Auto advance</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values/strings.xml:409]: Archiving, deletion, but also moving Message/Thread.
Are we modifying the Locos strings to take this into account ?
Has iOS also implemented the feature when moving ?

File: app/src/main/res/values-it/strings.xml [hunk 33]
```diff
@@ -390,10 +390,21 @@
...
+    <string name="settingsAutoAdvancePreviousThreadTitle">Filo conduttore precedente</string>
+    <string name="settingsAutoAdvanceTitle">Anticipo auto</string>
     <string name="settingsCancellationPeriodDescription">Avete la possibilità di annullare l’invio della vostra email fino a 30 secondi.</string>
     <string name="settingsCancellationPeriodTitle">Periodo di cancellazione dell’invio</string>
     <string name="settingsDataManagementDescription">È possibile personalizzare i dati di utilizzo e operativi che Infomaniak recupera per correggere i bug e migliorare l’applicazione. Entrambi gli strumenti sono ospitati e gestiti esclusivamente da noi.\n\nInfomaniak rispetta la vostra privacy: il codice sorgente di questa applicazione può essere verificato, le informazioni raccolte sono anonime e i vostri dati personali rimangono strettamente confidenziali e non vengono mai condivisi con terze parti.</string>
-    <string name="settingsDataManagementSourceCode">Vedere il codice sorgente</string>
+    <string name="settingsDataManagementSourceCode">Vedi il codice sorgente</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-it/strings.xml:407]: Remove this diff.

File: app/src/main/res/values-it/strings.xml [hunk 34]
```diff
@@ -403,7 +414,7 @@
     <string name="settingsDisabled">Disabili</string>
     <string name="settingsEnabled">Abilitato</string>
     <string name="settingsExternalContentDescription">Visualizzare il contenuto esterno</string>
-    <string name="settingsExternalContentTitle">contenuto esterno</string>
+    <string name="settingsExternalContentTitle">Contenuto esterno</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-it/strings.xml:417]: Remove this diff.

File: app/src/main/res/values-it/strings.xml [hunk 35]
```diff
@@ -415,7 +426,7 @@
     <string name="settingsMatomoDescription">Matomo è uno strumento di analisi ospitato e gestito esclusivamente da Infomaniak per capire come viene utilizzata l’applicazione.\n\nStudiando questi dati, il nostro team è in grado di migliorare continuamente l’interfaccia dell’applicazione.</string>
     <string name="settingsMatomoTitle">Matomo</string>
     <string name="settingsOptionAlways">Sempre</string>
-    <string name="settingsOptionAskMe">Chiedetemi</string>
+    <string name="settingsOptionAskMe">Chiedi</string>
```
Comment (KevinBoulongne) [inline app/src/main/res/values-it/strings.xml:429]: Remove this diff.

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 32 above)
Comment (NicolasBourdin88) [reply app/src/main/res/values/strings.xml:410]: Yes they implemented the feature when moving

File: app/src/main/res/values-es/strings.xml [hunk 36]
```diff
@@ -391,6 +391,17 @@
     <string name="settingsAiEngineDescription">Seleccione su asistente de redacción</string>
     <string name="settingsAppLock">Bloqueo de aplicaciones</string>
     <string name="settingsAuthorizeTracking">Autorizar el seguimiento</string>
+    <string name="settingsAutoAdvanceDescription">Seleccione qué mostrar cuando se archiva o elimina un mensaje</string>
+    <string name="settingsAutoAdvanceFollowingThreadDescription">Mostrar la siguiente conversación al archivar o borrar</string>
+    <string name="settingsAutoAdvanceFollowingThreadTitle">Siguiendo la conversación</string>
+    <string name="settingsAutoAdvanceListOfThreadsDescription">Mostrar lista de conversaciones al archivar o borrar</string>
```
Comment (LunarX) [inline app/src/main/res/values-es/strings.xml:397]: It's `Mostrar la lista...` you're missing a determinant. You have the same issue with the english translation

**Review by LunarX — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 32 above)
Comment (KevinBoulongne) [reply app/src/main/res/values/strings.xml:411]: The question was asked to the UI team, waiting for their answer.

File: app/src/main/res/values-es/strings.xml (hunk 36 above)
Comment (LunarX) [reply app/src/main/res/values-es/strings.xml]: You still have the issue with the english translation

**Review by LunarX — APPROVED** [review]

File: app/src/main/res/values/strings.xml (hunk 32 above)
Comment (KevinBoulongne) [reply app/src/main/res/values/strings.xml]: The decision was taken to merge as is, and to update the wording later.

File: app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt [hunk 37]
```diff
@@ -838,7 +846,6 @@ class MainViewModel @Inject constructor(
             snackbarManager.postValue(appContext.getString(snackbarTitle))
         }
     }
-    //endregion
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/MainViewModel.kt:841]: wat

**Review by KevinBoulongne — CHANGES_REQUESTED** [review]

**Review by KevinBoulongne — APPROVED** [review]
